processors/baggage/baggagetrace/processor.go

```diff
@@ -11,23 +11,36 @@ import (
 	"go.opentelemetry.io/otel/sdk/trace"
 )
 
+// BaggageKeyPredicate is a function that returns true if the baggage key should be added to the span.
+type BaggageKeyPredicate func(baggageKey string) bool
+
+// AllowAllBaggageKeys is a BaggageKeyPredicate that allows all baggage keys.
+var AllowAllBaggageKeys BaggageKeyPredicate = func(string) bool { return true }
+
 // SpanProcessor is a processing pipeline for spans in the trace signal.
-type SpanProcessor struct{}
+type SpanProcessor struct {
+	baggageKeyPredicate BaggageKeyPredicate
+}
 
 var _ trace.SpanProcessor = (*SpanProcessor)(nil)
 
 // New returns a new SpanProcessor.
 //
 // The Baggage span processor duplicates onto a span the attributes found
 // in Baggage in the parent context at the moment the span is started.
-func New() trace.SpanProcessor {
-	return &SpanProcessor{}
+// The predicate function is used to filter which baggage keys are added to the span.
```

Comment:
Might be useful to document the `AllowAllBaggageKeys` filter for discoverability

Reply:
It did have a docs entry, but because it also re-declared the type in the var, it thought it was another type definition. Fixed in [a8fbc83](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5619/commits/a8fbc837f2d3432755e90c7f2fa4c2bff3fcba6c)